proto/osmosis/gamm/v1beta1/pool.proto

```diff
@@ -150,3 +150,26 @@ message Pool {
     (gogoproto.nullable) = false
   ];
 }
+
+message TwapPair {
+  string tokenIn = 1;
+  string tokenOut = 2;
+  string priceCumulative = 3 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable) = false
+  ];
+  string spotPrice = 4 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Dec",
+    (gogoproto.nullable) = false
+  ];
+}
+
+message PoolTwapHistory {
+  google.protobuf.Timestamp timeStamp = 1 [
+    (gogoproto.stdtime) = true,
+    (gogoproto.nullable) = false,
+    (gogoproto.moretags) = "yaml:\"time_stamp\""
+  ];
+  uint64 poolId = 2;
+  repeated TwapPair twapPairs = 3;
```

Comment:
How are TwapPairs going to be sequenced, or will it be that everything that wants to look at a given token-pair will iterate over all pairs to find it?

Reply:
Also looking at this makes me think we should lower the max number of assets per pool from 8 to 5. 8 assets -> 28 pairs, 5 assets -> 10 pairs.

Reply:
How are TwapPairs going to be sequenced, or will it be that everything that wants to look at a given token-pair will iterate over all pairs to find it?- Yes basically it would hold info on all the possible twap pairs, and if changes are necessary for a specific pair, it would iterate over all the pairs. Do you think I could use a better algorithm / structure?

Reply:
That seems fine to me for now, it really depends on how many assets we expect to ever get per pool. If we only want like 8 assets max, then its not actually an issue to iterate over everything. If we want like 20 assets, then we need to rethink it (but we can also punt rethinking it to later)

cc @UnityChaos

Reply:
hmm I think it's fine for now, non-binary pools aren't getting much use yet, especially up to the 8 limit. We can revisit / optimize if that does become a concern tho.

Also for vaults I don't think we would be calculating TWAP on the vault itself, so no concern there.